src/Form/TfaLoginForm.php

```diff
@@ -70,20 +70,22 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
     // Similar to tfa_user_login() but not required to force user logout.
+    /** @var $account \Drupal\user\UserInterface */
+    $account = $this->userStorage->load($form_state->get('uid'));
 
-    if ($uid = $form_state->get('uid')) {
-      $account = $this->userStorage->load($uid);
-    }
-    else {
-      $account = user_load_by_name($form_state->get('name'));
-    }
 
-    if ($tfa = $this->tfaManager->getProcess($account)) {
-      if ($account->hasPermission('require tfa') && !$this->tfaManager->loginComplete($account) && !$tfa->ready()) {
+    //GetPlugin
+    //Pass to service functions.
+    /** @var  $tfaValidationPlugin \Drupal\tfa\Plugin\TfaValidationInterface */
+    $tfaValidationPlugin = $this->tfaValidationManager->getInstance(['uid' => $account->id()]);
+
+    //Setup TFA
+    if (isset($tfaValidationPlugin)) {
+      if ($account->hasPermission('require tfa') && !$this->loginComplete($account) && !$this->ready($tfaValidationPlugin)) {
```

Comment:
This condition is setting an error if TFA is trying to force the user to use TFA but they dont have it setup (probably meaning they can never do it). Currently they can only setup the globally configured TFAValidation type to use, but it has been talked about to allow them to choose from the enabled ones.